### DISCUSSION TITLE
Scene Filename Parser Improvement

### DISCUSSION THREAD

**Discussion by salamnkalalo** [stashapp/stash#3659] [Ideas]
Just started using `Stash` and I am loving it, but I have a slight issue with managing my file.
I am trying to parse file names into scene names and studio name in order to scrap them by name, but I found it very difficult using the  `Scene Filename Parser` tab.

For example I have the following file `/<STUDIO>/<ID> <ACTOR> -  <NAME> <TAG> .<EXT>` right now its impossible to select a specific studio name and also capture.
Another issue is that you can't control the outputs scene name using other captured tags, in my case I need to convert the file scene name to `<STUDIO>  <NAME>`.

A very simple solution could be to add RE support and file filtering support. A bit like the `pathParser` plugin but more ergonomic to use (no need to change local config files and add and remove tags).

Would love to know your thoughts.

**Comment by Flashy78** [comment]
I would suggest generating phashes and trying to match that first before worrying about searching by name.
https://docs.stashapp.cc/beginner-guides/guide-to-scraping/

The filename parser is definitely not fully featured, altough there was some talk about maybe in the future incorporating the `renameOnUpdate` plugin (not the old `renamer` plugin) into a first class feature. If that happens, then there would probably be code written that could easily be re-used to improve the filename parser in the ways you suggest.

**Reply by salamnkalalo** [reply]
I generated `phashes` for all my files but a lot of them aren't present on `stash-db` so I resorted to using `ThePornDB` which dose contain them.
I'll look into `renameOnUpdate` hopefully its better then `renamer`.
Glad to know you guys are aware of the situation.

**Reply by salamnkalalo** [reply]
After looking a bit into `renameOnUpdate` it seems its can only change the filename based on the scene and not the other way around. Am I missing anything?
EDIT:
My bad I tried using `pathParser`not `renamer`

**Comment by stg-annon** [comment]
I know it may not be an Ideal solution but I've found scrapers a much more reliable way to do this type of parsing, I think there is one in the community repo, if not I can post what I use, to use it in bulk you can use either the Tagger view or the Identify task in combination with the scraper

**Reply by salamnkalalo** [reply]
I tried using the `ThePornDB` but its very brittle because it doesn't use `phashes` for example adding the video ID before the name ruins the search results. any way of dealing with that?

**Reply by stg-annon** [reply]
if you use the stashdb endpoint they have that should do better at matching
